modules/core/src/main/scala/org/scalasteward/core/update/UpdateAlg.scala

```diff
@@ -37,38 +37,9 @@ final class UpdateAlg[F[_]](implicit
       dependency: Scope[Dependency],
       maxAge: Option[FiniteDuration]
   ): F[Option[Update.Single]] =
-    for {
-      existingVersions <- versionsCache.getVersions(dependency, maxAge)
-      maybeUpdate = maybeNewerVersions(dependency, existingVersions)
-        .map(vs => Update.Single(CrossDependency(dependency.value), vs.map(_.value)))
-      maybeUpdateOrRename <- maybeUpdate match {
-        case Some(update) => F.pure(Some(update))
-        case None =>
-          artifactMigrationsFinder.findUpdateWithRenamedArtifact(dependency.value) match {
-            case Some(artifactChange) => verifyVersion(dependency, artifactChange, maxAge)
-            case None                 => F.pure(None)
-          }
-      }
-    } yield maybeUpdateOrRename
-
-  private def verifyVersion(
-      dependency: Scope[Dependency],
-      artifactChange: ArtifactChange,
-      maxAge: Option[FiniteDuration]
-  ): F[Option[Update.Single]] =
-    versionsCache
-      .getVersions(migratedDependency(dependency, artifactChange), maxAge)
-      .map { existingVersions =>
-        maybeNewerVersions(dependency, existingVersions)
-          .map(newerVersions =>
-            Update.Single(
-              CrossDependency(dependency.value),
-              newerVersions.map(_.value),
-              Some(artifactChange.groupIdAfter),
-              Some(artifactChange.artifactIdAfter)
-            )
-          )
-      }
+    findUpdateWithoutMigration(dependency, maxAge)
+      .orElse(findUpdateWithMigration(dependency, maxAge))
+      .value
```

Comment:
I really like the abstraction you have here now, easy to grasp what is going on.